app/controllers/problems_controller.rb

```diff
@@ -12,10 +12,30 @@ def index
   end
 
   def create
-    problem = Problem.new(create_params)
-    if problem.save!
-      render status: :created, json: problem
+    problem = Problem.create!(create_params)
+
+    ActiveRecord::Base.transaction do
+      uploaded_file = params[:file]
```

Comment:
Can we impose a size limit? I know we havent done that yet, but we should. Ideally it would be stored in a constants or config file.